src/lib.rs

```diff
@@ -95,13 +122,27 @@ fn process(args: ProcessingArgs) -> Result<()> {
                     "Skipping already processed derivation: {}",
                     build_input.attribute_path.to_string()
                 );
+
+                // Inform calling thread that the derivation was skipped if
+                // requested.
+                if let Some(message_tx) = &args.message_tx {
+                    message_tx
+                        .send(message::Message {
+                            status: message::Status::Skipped,
+                            id: rayon::current_thread_index().unwrap(),
+                            path: build_input.attribute_path.clone(),
+                        })
+                        .unwrap();
+                }
```

Comment:
As those 3 block code look very similar, what do you think of pulling the common code in a local function, something like:

```rust
fn inform(msg_tx: Option<mspc::Sender<_>>, status: message::Status) {
  // boilerplate
}

Reply:
That's a good idea, thank you <3